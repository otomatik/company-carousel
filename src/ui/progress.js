import React from 'react';
import styled from 'styled-components';

export const Progress = ({ percentage }) => (
  <Container>
    <Fill percentage={percentage} />
  </Container>
);

const Container = styled.div`
  flex: 1;
  height: 4px;
  background: #333;
`;

const Fill = styled.div`
  width: ${props => props.percentage}%;
  transition: width 300ms ease-in-out;
  height: 100%;
  background: #50805d;
`;
