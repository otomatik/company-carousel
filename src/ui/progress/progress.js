import React from 'react';
import styled from 'styled-components';

export const Progress = ({ percentage }) => (
  <Container>
    <Fill percentage={percentage} />
  </Container>
);

const Container = styled.div`
  flex: 1;
  height: 4px;
  background: #333;
`;

const Fill = styled.div`
  height: 100%;
  width: ${props => props.percentage}%;
  transition: width 300ms ease-in-out;
  background: rgb(0, 194, 154);
`;
