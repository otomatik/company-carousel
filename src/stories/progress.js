import { storiesOf } from '@storybook/react';
import { number, withKnobs } from '@storybook/addon-knobs';
import { Progress } from '../ui/progress';
import React from 'react';

export const progressStories = storiesOf('Progress', module)
  .addDecorator(withKnobs)
  .add('horizontal', () => <Progress percentage={number('percentage', 50)} />);
