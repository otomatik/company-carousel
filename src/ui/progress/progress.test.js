import { Progress } from './progress';
import { shallow } from 'enzyme';
import React from 'react';

describe('Progress', () => {
  it('should match snapshot', () => {
    // when
    const progress = shallow(<Progress percentage={42} />);

    // then
    expect(progress).toMatchSnapshot();
  });
});
