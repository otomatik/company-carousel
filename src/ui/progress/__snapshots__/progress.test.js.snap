// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should match snapshot 1`] = `
<styled.div>
  <styled.div
    percentage={42}
  />
</styled.div>
`;
